Refactor PlanEval and humanoid controller code

siyuanfeng-tri opened the pull request:
1. Moved the basic PlanEval constructs from examples/QPInverseDynamicsForHuamnoids to systems/controllers/plan_eval. 
2. Moved all humanoid related code from examples/QPInverseDynamicsForHuamnoids to examples/humanoid_controller. This can be promoted to drake/legged_locomotion someday. 
Added a readme there 
3. Replaced the examples/QPInverseDynamicsForHuamnoids folder with examples/qp_inverse_dynamics, which now contains a minimal example of using plan eval + qp inverse dynamics to write a joint space impedance controller. This is functionally equivalent to systems/controller/inverse_dynamics_controller.cc in the unconstrained case (not hitting joint torque / acceleration limit etc).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/7109)
<!-- Reviewable:end -->

siyuanfeng-tri commented:
+@ericcousineau-tri for feature review. thanks!

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-:-KuzeyFdxwtGc8Qp0JF6:b4k5yms)*
<!-- Sent from Reviewable.io -->

EricCousineau-TRI commented:
First pass done, PTAL when you have the chance. Thanks!

---

Reviewed 84 of 84 files at r1.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/examples/humanoid_controller/humanoid_controller.h, line 59 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzi2fQKEX3QffeKd0F:-Kuzi2fRjfEnnZiNobQd:b7umt8x) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/humanoid_controller.h#L59)):*
> ```objc
>     qp_con->set_name("qp_con");
> 
>     AtlasCommandTranslatorSystem* joint_con =
> ```

Is there a way that the name `AtlasCommandTranslatorSystem` can be generalized, so that it fits in `HumanoidController`?

---

*[drake/examples/humanoid_controller/humanoid_status_translator_system.h, line 52 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzi_lN1XiK4YTcMinw:-Kuzi_lN1XiK4YTcMinx:b-tdj452) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/humanoid_status_translator_system.h#L52)):*
> ```objc
>   void CalcHumanoidStatus(
>       const systems::Context<double>& context,
>       systems::controllers::qp_inverse_dynamics::RobotKinematicState<double>*
> ```

BTW Could the awkward line break be fixed with a `typedef`?

---

*[drake/examples/humanoid_controller/README.md, line 1 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzjY1wEvo0eEaMrdGJ:-KuzjY1xfxcozrDkMyrl:b53y5j) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/README.md#L1)):*
> ```Markdown
> A demo of valkyrie executing a manipulation plan while actively maintaining
> ```

Would you be able to reformat this with Markdown?

Here's my attempt:
https://github.com/EricCousineau-TRI/drake/commit/87bf87da7d50dfd82a51b130d37dfd9e2d190f65

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_controller.cc, line 23 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzkOlcB5arvH4fO1p7:-KuzkOlcB5arvH4fO1p8:bje1wqn) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/qp_inverse_dynamics/manipulator_move_joint_controller.cc#L23)):*
> ```cpp
>     RobotKinematicStateTranslatorSystem;
> 
> ManipulatorMoveJointController::ManipulatorMoveJointController(
> ```

Might there be a better name than "MoveJointController"? Something like "JointSpaceController"?

(It still loses the fact that it's inverse dynamics, but I understand it'd be hard to capture that in a not-painful-to-use name.)

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzkaPVwZwlsqKGl6hz:-KuzkaPWUA5n8TQo2k4Y:b51gljg) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc#L25)):*
> ```cpp
>   DRAKE_DEMAND(get_robot().get_num_velocities() ==
>                get_robot().get_num_positions());
>   // Skips the first body(world), which doesn't have a joint.
> ```

BTW Space in between: "body (world)"

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc, line 27 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzkgnRkFMjVtVrjnkS:-KuzkgnRkFMjVtVrjnkT:b-138kac) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc#L27)):*
> ```cpp
>   // Skips the first body(world), which doesn't have a joint.
>   for (size_t i = 1; i < robot->bodies.size(); i++) {
>     DRAKE_DEMAND(!robot->bodies[i]->getJoint().is_floating());
> ```

BTW Is there a way to delegate this logic to check if it's actuator, rather than rejecting floating base joints?
(Looking at RigidBodyTree, looks like that's not the case, but just curious)

---

*[drake/systems/controllers/qp_inverse_dynamics/robot_kinematic_state_translator_system.h, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzmk5dFyQSVRszmerJ:-Kuzmk5dFyQSVRszmerK:bhu1h6l) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/systems/controllers/qp_inverse_dynamics/robot_kinematic_state_translator_system.h#L13)):*
> ```objc
> 
> /**
>  * A simple translator class from a state vector to RobotKinematicState.
> ```

BTW nit Doxygen should be declarative ("Provides translation from...").

---

*[drake/systems/controllers/qp_inverse_dynamics/robot_kinematic_state_translator_system.h, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzmcLs6jkFa6vhccdS:-KuzmcLtbEkOYE6fgzFb:b-lp5cfm) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/systems/controllers/qp_inverse_dynamics/robot_kinematic_state_translator_system.h#L42)):*
> ```objc
> 
>   /**
>    * Returns the input port for a state vector.
> ```

BTW nit grammar, "~a~ the state vector"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-:-Kuzn84rHi337zCmOe9G:b-o19p0v)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
Review status: 46 of 53 files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/examples/humanoid_controller/humanoid_controller.h, line 59 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzi2fQKEX3QffeKd0F:-Kv-BuCeRYQzEG1_0lOu:b-sj7wmw) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/humanoid_controller.h#L59)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

Is there a way that the name `AtlasCommandTranslatorSystem` can be generalized, so that it fits in `HumanoidController`?
</blockquote></details>

atlas_command was the lcm message's type which is somewhat not generalizable. I kinda want to keep this name so i know what's it translating to. 

---

*[drake/examples/humanoid_controller/humanoid_status_translator_system.h, line 52 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzi_lN1XiK4YTcMinw:-Kv-1yBaJxIC8Mg6dWhX:bwcawmb) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/humanoid_status_translator_system.h#L52)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW Could the awkward line break be fixed with a `typedef`?
</blockquote></details>

the typedef would be pretty long (and awkward) as well? i don't really like adding typedefs unless really necessary. 

---

*[drake/examples/humanoid_controller/README.md, line 1 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzjY1wEvo0eEaMrdGJ:-Kv-2L3RYUYxtdIHI1TT:b-9qya5b) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/README.md#L1)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

Would you be able to reformat this with Markdown?

Here's my attempt:
https://github.com/EricCousineau-TRI/drake/commit/87bf87da7d50dfd82a51b130d37dfd9e2d190f65
</blockquote></details>

wow thanks. 

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzkaPVwZwlsqKGl6hz:-Kv-4A1Nu_mNJS9Eqrp4:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc#L25)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW Space in between: "body (world)"
</blockquote></details>

Done.

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc, line 27 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzkgnRkFMjVtVrjnkS:-Kv-3C8xZEVPG2K4ENVw:b-roaqw8) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc#L27)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW Is there a way to delegate this logic to check if it's actuator, rather than rejecting floating base joints?
(Looking at RigidBodyTree, looks like that's not the case, but just curious)
</blockquote></details>

You mention an interesting question about under actuation. The plan eval part doesn't really care about actuation, the qp part will still try to achieve the desired vd as closely as possible despite under actuation. 
The problem I am trying to rule out is q* - q. for a floating joint, - needs to work with rotation properly. 

---

*[drake/systems/controllers/qp_inverse_dynamics/robot_kinematic_state_translator_system.h, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzmk5dFyQSVRszmerJ:-Kv-5IXTvw7r83N0M48D:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/systems/controllers/qp_inverse_dynamics/robot_kinematic_state_translator_system.h#L13)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW nit Doxygen should be declarative ("Provides translation from...").
</blockquote></details>

Done.

---

*[drake/systems/controllers/qp_inverse_dynamics/robot_kinematic_state_translator_system.h, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzmcLs6jkFa6vhccdS:-Kv-5B7qlIsbiPWKOZAH:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/systems/controllers/qp_inverse_dynamics/robot_kinematic_state_translator_system.h#L42)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW nit grammar, "~a~ the state vector"
</blockquote></details>

Done.

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_controller.cc, line 23 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzkOlcB5arvH4fO1p7:-Kv-5QfIZUPcFrxRkcQc:bs7wumx) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/qp_inverse_dynamics/manipulator_move_joint_controller.cc#L23)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

Might there be a better name than "MoveJointController"? Something like "JointSpaceController"?

(It still loses the fact that it's inverse dynamics, but I understand it'd be hard to capture that in a not-painful-to-use name.)
</blockquote></details>

Done. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109)*
<!-- Sent from Reviewable.io -->

EricCousineau-TRI commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 87 files at r1, 10 of 10 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/examples/humanoid_controller/humanoid_controller.h, line 59 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzi2fQKEX3QffeKd0F:-Kv2oGYkZ3gv6WVPJ_Gf:bfwldho) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/humanoid_controller.h#L59)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

atlas_command was the lcm message's type which is somewhat not generalizable. I kinda want to keep this name so i know what's it translating to. 
</blockquote></details>

Sounds good to me.
Could you add a comment here mentioning this, maybe making it a TODO to have something more generalized?

---

*[drake/examples/humanoid_controller/humanoid_status_translator_system.h, line 52 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzi_lN1XiK4YTcMinw:-Kv2oTNalYifuUXNUqwY:brxp9sq) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/humanoid_status_translator_system.h#L52)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

the typedef would be pretty long (and awkward) as well? i don't really like adding typedefs unless really necessary. 
</blockquote></details>

Sounds good.

---

*[drake/examples/humanoid_controller/README.md, line 1 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzjY1wEvo0eEaMrdGJ:-Kv2oQXEw93UzCYM6suA:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/README.md#L1)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

wow thanks. 
</blockquote></details>

Done.

---

*[drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc, line 34 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv2odwOLHz7ddOAid2B:-Kv2odwOLHz7ddOAid2C:b-afqlo) ([raw file](https://github.com/robotlocomotion/drake/blob/765655a99e73ae5aa9ccf4c8201857438f6d7016/drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc#L34)):*
> ```cpp
>   const RigidBodyTree<double>& robot = *robot_for_control_;
> 
>   this->set_name("ManipulatorJointSpaceController");
> ```

Is it standard practice to set names within a system, without it being a parameter?
Might this cause a conflict if two controllers are instantiated within the same diagram?

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc, line 27 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-KuzkgnRkFMjVtVrjnkS:-Kv2oyXjYSg3mJyeTy_N:b-1lje90) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.cc#L27)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

You mention an interesting question about under actuation. The plan eval part doesn't really care about actuation, the qp part will still try to achieve the desired vd as closely as possible despite under actuation. 
The problem I am trying to rule out is q* - q. for a floating joint, - needs to work with rotation properly. 
</blockquote></details>

Gotcha, thanks for the explanation!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-:-Kv2p6RnMIoeGjElpozf:bnfp4nl)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
+@seancurtis-tri for platform review. thanks! 

---

Review status: 52 of 53 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/examples/humanoid_controller/humanoid_controller.h, line 59 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kuzi2fQKEX3QffeKd0F:-Kv3A4y3nW0-ejJ7wSw8:b-rnrapz) ([raw file](https://github.com/robotlocomotion/drake/blob/6bb1a107815ee765e3f3be0386b142d70a0671a0/drake/examples/humanoid_controller/humanoid_controller.h#L59)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

Sounds good to me.
Could you add a comment here mentioning this, maybe making it a TODO to have something more generalized?
</blockquote></details>

done. 

---

*[drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc, line 34 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv2odwOLHz7ddOAid2B:-Kv3ABs7uW8UMcCK2FrJ:b-77503m) ([raw file](https://github.com/robotlocomotion/drake/blob/765655a99e73ae5aa9ccf4c8201857438f6d7016/drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc#L34)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

Is it standard practice to set names within a system, without it being a parameter?
Might this cause a conflict if two controllers are instantiated within the same diagram?
</blockquote></details>

hmm. That's a good question, I am leaving it to platform. 
Once upon a time, it throws exceptions if no name is specified. A lot of these set_names came right after that. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-:-Kv3Ar87ELkz2l7QqESd:b-xi6woz)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
On the whole, <img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>. A typo and some questions.

---

Reviewed 76 of 87 files at r1, 10 of 10 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/examples/humanoid_controller/README.md, line 11 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv48Y2T_JArJ78Lm0SV:-Kv48Y2T_JArJ78Lm0SW:b-4wud1l) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/README.md#L11)):*
BTW Grammar here; not sure what it should be.  

> ...and uses it to offset the right should pitch...

Perhaps "should" should be cut?  Looking at other files, I now suspect: s/should/shoulder.

---

*[drake/examples/humanoid_controller/README.md, line 24 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv48pJJDxTAneHjJZ9H:-Kv48pJJDxTAneHjJZ9I:bjnu4dm) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/README.md#L24)):*
> ```Markdown
> 1. Launch the visualizer before the simulator:
> 
>     `$ ./bazel-bin/tools/drake_visualizer`
> ```

FYI This assumes that `bazel build //tools:drake_visualizer` has previously been run. Similar for all subsequent commands. Perhaps a note about building the targets?

---

*[drake/examples/humanoid_controller/send_manipulation_plan_demo.cc, line 76 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv49FUhUuUQXlLPGx0o:-Kv49FUhUuUQXlLPGx0p:bwat43p) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/send_manipulation_plan_demo.cc#L76)):*
> ```cpp
> 
> int main(int argc, char** argv) {
>   double right_shoulder_pitch_offset = 0;
> ```

FYI Any particular reason you're not using gflags?

---

*[drake/examples/humanoid_controller/test/atlas_command_translator_system_test.cc, line 6 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv4BtbXmWXra8_OKj2K:-Kv4BtbXmWXra8_OKj2L:b-7s58ht) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/test/atlas_command_translator_system_test.cc#L6)):*
> ```cpp
> 
> #include <gtest/gtest.h>
> #include "bot_core/atlas_command_t.hpp"
> ```

FYI While not explicitly part of the style guide, this use of double quotes is definitely exceptional in drake.

---

*[drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc, line 34 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv2odwOLHz7ddOAid2B:-Kv4Eo6YdCK8vT6uBJpi:b-f96j41) ([raw file](https://github.com/robotlocomotion/drake/blob/765655a99e73ae5aa9ccf4c8201857438f6d7016/drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc#L34)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

hmm. That's a good question, I am leaving it to platform. 
Once upon a time, it throws exceptions if no name is specified. A lot of these set_names came right after that. 
</blockquote></details>

If you omit the call entirely, instantiation via the `DiagramBuilder` will introduce a name that includes the class name. So, setting it here may be redundant (and lead to name collisions as @ericcousineau-tri alluded. On the other hand, names are purely decorative -- we don't have any functionality that *depends* on the uniqueness...

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.h, line 22 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv4GauuGsUaTDvKeKn4:-Kv4GauuGsUaTDvKeKn5:bhcsfaq) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.h#L22)):*
> ```objc
>  * Assuming no external contacts, the control law is:
>  * <pre>
>  *   vd_output = Kp(q* - q) + Kd(v* - v) + vd*,
> ```

FYI This gave me a moment of pause. Any way we can replace `vd` with `vdt` or `v̇`?

---

*[drake/systems/controllers/qp_inverse_dynamics/test/qp_output_translator_system_test.cc, line 27 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv4NLmo6clAzUAs5MCF:-Kv4NLmo6clAzUAs5MCG:bh3pqq2) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/systems/controllers/qp_inverse_dynamics/test/qp_output_translator_system_test.cc#L27)):*
> ```cpp
>   parsers::urdf::AddModelInstanceFromUrdfFile(
>       FindResourceOrThrow(
>         "drake/examples/valkyrie/urdf/urdf/"
> ```

FYI Does this indentation survive clang format?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-:-Kv4NjJkkee16yTQn85V:b43hoo5)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/examples/humanoid_controller/README.md, line 11 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv48Y2T_JArJ78Lm0SV:-Kv4Ql4ASHxDwKR-KLUX:bze84a1) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/README.md#L11)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW Grammar here; not sure what it should be.  

> ...and uses it to offset the right should pitch...

Perhaps "should" should be cut?  Looking at other files, I now suspect: s/should/shoulder.
</blockquote></details>

opps shoulder. 

---

*[drake/examples/humanoid_controller/README.md, line 24 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv48pJJDxTAneHjJZ9H:-Kv4U95SigmG5DEx3As4:b9sku7z) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/README.md#L24)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI This assumes that `bazel build //tools:drake_visualizer` has previously been run. Similar for all subsequent commands. Perhaps a note about building the targets?
</blockquote></details>

added a to build. 

---

*[drake/examples/humanoid_controller/send_manipulation_plan_demo.cc, line 76 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv49FUhUuUQXlLPGx0o:-Kv4UE2sT0n36ASPtYDI:b-sty6e2) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/send_manipulation_plan_demo.cc#L76)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI Any particular reason you're not using gflags?
</blockquote></details>

because i was lazy.. switching to gflags. 

---

*[drake/examples/humanoid_controller/test/atlas_command_translator_system_test.cc, line 6 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv4BtbXmWXra8_OKj2K:-Kv4VHFeDqQnpfuivmf2:b-2wplyn) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/test/atlas_command_translator_system_test.cc#L6)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI While not explicitly part of the style guide, this use of double quotes is definitely exceptional in drake.
</blockquote></details>

what should i be using <>?

---

*[drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc, line 34 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv2odwOLHz7ddOAid2B:-Kv4hln9Xndyc8F6Ys4C:be9jziy) ([raw file](https://github.com/robotlocomotion/drake/blob/765655a99e73ae5aa9ccf4c8201857438f6d7016/drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc#L34)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

If you omit the call entirely, instantiation via the `DiagramBuilder` will introduce a name that includes the class name. So, setting it here may be redundant (and lead to name collisions as @ericcousineau-tri alluded. On the other hand, names are purely decorative -- we don't have any functionality that *depends* on the uniqueness...
</blockquote></details>

ok. so what should i do then? 
regardless of the answer, i'd rather do that in a separate path, because there are a fair number of these set_names everywhere. too big for this pr. 

---

*[drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.h, line 22 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv4GauuGsUaTDvKeKn4:-Kv4h546fefFNPHU3C1R:ba8wv3n) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/qp_inverse_dynamics/manipulator_move_joint_plan_eval_system.h#L22)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI This gave me a moment of pause. Any way we can replace `vd` with `vdt` or `v̇`?
</blockquote></details>

switched vdot.  

---

*[drake/systems/controllers/qp_inverse_dynamics/test/qp_output_translator_system_test.cc, line 27 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv4NLmo6clAzUAs5MCF:-Kv4hWuDCno7kXZJ5Vsw:bb7l5xa) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/systems/controllers/qp_inverse_dynamics/test/qp_output_translator_system_test.cc#L27)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI Does this indentation survive clang format?
</blockquote></details>

nah.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
see #7115 for comments on the naming. 

---

Review status: 48 of 53 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-:-Kv581sZ4syGATLNGFTs:b-fc2wca)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Reviewed 2 of 87 files at r1, 5 of 5 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[drake/examples/humanoid_controller/test/atlas_command_translator_system_test.cc, line 6 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv4BtbXmWXra8_OKj2K:-Kv7hmoLBYIsbDwxsoHa:b-zdhbe5) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/test/atlas_command_translator_system_test.cc#L6)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

what should i be using <>?
</blockquote></details>

Yeah, the general sense I get (from sampling and conversation) is that we reserve quotes for drake code. Again, nothing explicit in the style guide (as far as I'm aware), but... And as long as you've got CI issues that need to be addressed...

---

*[drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc, line 34 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv2odwOLHz7ddOAid2B:-Kv7i3MDtBTWgy0Akith:bugnbw8) ([raw file](https://github.com/robotlocomotion/drake/blob/765655a99e73ae5aa9ccf4c8201857438f6d7016/drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc#L34)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

ok. so what should i do then? 
regardless of the answer, i'd rather do that in a separate path, because there are a fair number of these set_names everywhere. too big for this pr. 
</blockquote></details>

1) Deferring to subsequent PR is acceptable. (How about a TODO at least on *this* name.
2) Names in systems are generally a sloppy concept in Drake that need to be cleaned up. You might defer "fixing" this until we know what the fix is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
Review status: 47 of 53 files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/examples/humanoid_controller/test/atlas_command_translator_system_test.cc, line 6 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv4BtbXmWXra8_OKj2K:-Kv7lBi4sRIaIN3nMoXa:b-7gqk6r) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/test/atlas_command_translator_system_test.cc#L6)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

Yeah, the general sense I get (from sampling and conversation) is that we reserve quotes for drake code. Again, nothing explicit in the style guide (as far as I'm aware), but... And as long as you've got CI issues that need to be addressed...
</blockquote></details>

well. that's the first time i have heard about this. i have no opinions either way.. 
fixed the lint thing ci thing. 

---

*[drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc, line 34 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv2odwOLHz7ddOAid2B:-Kv7nkVm7KeC9x5pBNlq:b-powww0) ([raw file](https://github.com/robotlocomotion/drake/blob/765655a99e73ae5aa9ccf4c8201857438f6d7016/drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc#L34)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

1) Deferring to subsequent PR is acceptable. (How about a TODO at least on *this* name.
2) Names in systems are generally a sloppy concept in Drake that need to be cleaned up. You might defer "fixing" this until we know what the fix is.
</blockquote></details>

i just rmed these, i don't think they are any more intelligent names. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
*[drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc, line 34 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv2odwOLHz7ddOAid2B:-Kv7nqmoqvgogGQ4smw0:bbxz2bv) ([raw file](https://github.com/robotlocomotion/drake/blob/765655a99e73ae5aa9ccf4c8201857438f6d7016/drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc#L34)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

i just rmed these, i don't think they are any more intelligent names. 
</blockquote></details>

i will do another more broad scope pr once we know what to do about set_name in general. (there are lots of these in iiwa)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
(You weren't kidding about lots of names).

---

Reviewed 6 of 87 files at r1, 2 of 10 files at r2, 6 of 6 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/examples/humanoid_controller/humanoid_controller.h, line 48 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv7oO2UD9sfCiyo6b2P:-Kv7oO2UD9sfCiyo6b2Q:b-tc5di7) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/humanoid_controller.h#L48)):*
> ```objc
>         builder.AddSystem(std::make_unique<RobotStateMsgToHumanoidStatusSystem>(
>             robot_.get(), alias_group_path));
>     msg_to_humanoid_status->set_name("msg_to_humanoid_status");
> ```

FYI For the record, I don't think these are as bad as the ones that call `this->set_name()`. Still, if killin' 'em all works for you. I'm good.

---

*[drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc, line 34 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv2odwOLHz7ddOAid2B:-Kv7oexTA3x1go2u8teR:bt9gdla) ([raw file](https://github.com/robotlocomotion/drake/blob/765655a99e73ae5aa9ccf4c8201857438f6d7016/drake/examples/qp_inverse_dynamics/manipulator_joint_space_controller.cc#L34)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

i will do another more broad scope pr once we know what to do about set_name in general. (there are lots of these in iiwa)
</blockquote></details>

I'd went through this PR before reading through yesterday's final naming debate. :) Too bad I hadn't done it in the other order. @ericcousineau-tri I believe you're still blocking on this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-:-Kv7ovas_eDfiWQkX7ul:b-8fgbdj)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/examples/humanoid_controller/humanoid_controller.h, line 48 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv7oO2UD9sfCiyo6b2P:-Kv7pjCuid_IBCmD85Yb:b-ufnpi1) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/humanoid_controller.h#L48)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI For the record, I don't think these are as bad as the ones that call `this->set_name()`. Still, if killin' 'em all works for you. I'm good.
</blockquote></details>

kill them all!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/examples/humanoid_controller/humanoid_controller.h, line 48 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7109#-Kv7oO2UD9sfCiyo6b2P:-Kv7q7qSmdXJLgX1sDwb:b-8etbxt) ([raw file](https://github.com/robotlocomotion/drake/blob/f33502e9b7d99dc743d768f904a318acbd3d3eb5/drake/examples/humanoid_controller/humanoid_controller.h#L48)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

kill them all!
</blockquote></details>

:gun:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7109)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
@siyuanfeng-tri This PR added (renamed) a file drake/systems/controllers/plan_eval/manipulator_move_end_effector_plan.h and .cc, but nothing compiles that.  Do you want me remove those sources, or add a cc_library rule for them?
